### DISCUSSION TITLE
May I know why doesn't marimo notebooks stores cell outputs on restart?

### DISCUSSION THREAD

**Discussion by roberthobblebottom** [marimo-team/marimo#600] [Q&A] [answered]
I can disable autorun but I kind of want the outputs to remain there. The alternative right now is to copy and paste the output in markdown.  
  
edit: how about option to clear outputs of particular cells, will marimo have that?

Thank you for creating marimo

**Comment by akshayka** [comment, marked as answer]
There's an issue to persist outputs across restarts; we plan to implement this. https://github.com/marimo-team/marimo/issues/541

We don't plan to add outputs to the Python file. Removing outputs from the notebook file makes it easier to git version, but I understand it makes the file less of a record of your work. You can export your notebook with outputs to HTML after you are done working.

We don't plan to have the option to clear outputs of arbitrary cells, but you can clear the output of the currently running cell with https://docs.marimo.io/api/outputs.html#marimo.output.clear

**Reply by roberthobblebottom** [reply]
I see, it is easier for versioning but not as good for notetaking, I have not thought about the html export. I will try this out.

**Reply by roberthobblebottom** [reply]
maybe if there is a way to turn the output into a new cell with mo.md() markdown? Maybe call it mo.log() or something.   
  
or these output logs and markdowns be in a different file so that people could just gitignore them but when editing or running both files will be used?
  
Or maybe this suggestion is flawed in ways I have not think off haha. Thanks again :)
